src/tester.rs

```diff
@@ -128,6 +130,25 @@ impl LangTester {
         self
     }
 
+    /// If set, defines what top-level lines will be treated as comments if they begin with
```

Comment:
top-level is confusing me a bit here.

If you look at a typical lang_test, it looks like:
```
// Compiler:
//   stdout:
//     match me!
```

And now we are adding the ability to have an inner comment, so a test can look like:
```
// Compiler:
//   stdout:
//     # flibble
//     match me!
```

So when you say "defines what top-level lines will be treated as comments", I'm thinking of the top-level `//` comments, not the `#' ones.

Do you see my concern?

Reply:
If you can think of a better phrasing, I'm all ears, but I think this comes under the heading of "this is how lang_tester works and you can't avoid knowing a bit about how a) your language parses comments b) your runner extracts test c) how lang_tester parses things".

Reply:
I've just realised something else. By top-level, you mean it must appear at the same level as (e.g.) `Compiler:`.

So in fact my example above is invalid. But should it be? I've often wanted to comment why I'm matching something, so perhaps comments should be allowed anywhere, and then all of this "top-level" talk disappears?

Reply:
It's true that you might accidentally comment something you want to match.

I'll leave the decision up to you. I'm aware that I've been pedantic/annoying enough for one PR!

Reply:
On reflection I agree that generalising this is probably a good idea. Try 2223188.